include/swift/AST/Types.h

```diff
@@ -2282,7 +2282,35 @@ getSILFunctionLanguage(SILFunctionTypeRepresentation rep) {
 
   llvm_unreachable("Unhandled SILFunctionTypeRepresentation in switch.");
 }
-
+  
+/// A call argument or parameter.
+struct CallArgParam {
+  /// The type of the argument or parameter. For a variadic parameter,
+  /// this is the element type.
+  Type Ty;
+  
+  // The label associated with the argument or parameter, if any.
+  Identifier Label;
+  
+  /// Whether the parameter has a default argument.  Not valid for arguments.
+  bool HasDefaultArgument = false;
```

Comment:
It doesn't belong in the type, though, and we don't want it to look like it has meaning there. I think `CallArgParam` needs to stay where it is, and we can introduce a `AnyFunctionType::Param` (or similar) that only captures what's actually in the type system.